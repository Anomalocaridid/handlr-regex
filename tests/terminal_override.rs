use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
use std::process::Command;

/// Helper function to test terminal output detection
fn test_terminal_output(terminal_output: bool) -> Command {
    let mut cmd = Command::new(get_cargo_bin("handlr"));
    cmd.arg(format!("--force-terminal-output={}", terminal_output))
        .arg("-vvv") // Maximum verbosity
        .arg("--disable-notifications") // Not much point showing these in tests
        .arg("mime")
        .arg("./assets");
    cmd
}

/// Helper function for insta settings
fn timestamp_filter() -> Vec<(&'static str, &'static str)> {
    vec![(r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d*\.\d*Z", "[TIMESTAMP]")]
}

#[test]
fn terminal_output_tests_force_true() {
    insta::with_settings!(
        {
            filters => timestamp_filter()
        },
        { assert_cmd_snapshot!(test_terminal_output(true)) }
    )
}

#[test]
fn terminal_output_tests_force_false() {
    insta::with_settings!(
        {
            filters => timestamp_filter()
        },
        { assert_cmd_snapshot!(test_terminal_output(false)) }
    )
}
