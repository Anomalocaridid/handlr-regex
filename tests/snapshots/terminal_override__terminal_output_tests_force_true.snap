---
source: tests/terminal_override.rs
info:
  program: handlr
  args:
    - "--force-terminal-output=true"
    - "-vvv"
    - "--disable-notifications"
    - mime
    - "./assets"
---
success: true
exit_code: 0
----- stdout -----
┌──────────┬────────────────────────┐
│[37m [39m[37mpath[39m[37m    [39m[37m [39m│[37m [39m[37mmime[39m[37m                  [39m[37m [39m│
├──────────┼────────────────────────┤
│[40m [49m[40m./assets[49m[40m [49m│[40m [49m[40mapplication/x-zerosize[49m[40m [49m│
└──────────┴────────────────────────┘

----- stderr -----
[2m[TIMESTAMP][0m [34mDEBUG[0m [2mhandlr[0m[2m:[0m Interactive terminal detected: true
[2m[TIMESTAMP][0m [32m INFO[0m [2mhandlr::common::path[0m[2m:[0m Printing mime information for paths: ["./assets"]
[2m[TIMESTAMP][0m [34mDEBUG[0m [2mhandlr::common::path[0m[2m:[0m JSON output: false
[2m[TIMESTAMP][0m [32m INFO[0m [2mhandlr::common::path[0m[2m:[0m Finished printing mime information
